python/rmm/_lib/memory_resource.pyx

```diff
@@ -299,6 +300,10 @@ cdef class PoolMemoryResource(UpstreamResourceAdaptor):
         """
         pass
 
+    def pool_size(self):
+        return (<pool_memory_resource[device_memory_resource]*>(
+            self.c_obj.get()))[0].pool_size()
```

Comment:
This feels like a lot to cram into one line. Would it make sense to break this up a bit by assigning to variables?

Reply:
I tried to clean this up a bit. Unfortunately, unlike C++, I can't create a type alias for `pool_memory_resource[device_memory_resource]`, so this is still a bit awkward looking.

Reply:
(Cython doesn't let  you `ctypedef` inside a function)

Reply:
The change looks much better! Thanks Ashwin 😄

Interesting guessing you tried `ctypedef`? Did you try `DEF`? In any event this seems less critical at least from my perspective